src/log/normal.c

```diff
@@ -431,10 +431,12 @@ void normal_log_message(enum criterion_severity severity, const char *msg)
     char *saveptr   = NULL;
     char *line      = strtok_r(dup, "\n", &saveptr);
 
-    do {
-        if (*line != '\0')
-            criterion_log_noformat(severity, line);
-    } while ((line = strtok_r(NULL, "\n", &saveptr)));
+    if (line) {
+        do {
+            if (*line != '\0')
+                criterion_log_noformat(severity, line);
+        } while ((line = strtok_r(NULL, "\n", &saveptr)));
+    }
```

Comment:
I think I'd prefer if we just short-circuited the logic at the very beginning of normal_log_message -- that way, if we have an empty string, we don't bother going through the malloc/copy/do-nothing/free cycle.